File: code/java/剑指offer54.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int kthLargest(TreeNode root, int k) {
        List<Integer> res = new ArrayList<>();
        Stack<TreeNode> stack = new Stack<>();
        while(true){
            while(root!=null){
                stack.push(root);
                root = root.left;
            }
            if(stack.isEmpty()) break;
            root = stack.pop();
            res.add(root.val);
            if(root.right!=null){
                root = root.right;
            }else{
                root = null;
            }
        }
        //System.out.print(res);
        return res.get(res.size()-k);
    }
}

File: code/python/剑指offer54.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def kthLargest(self, root: TreeNode, k: int) -> int:
        c = 0
        stack = []
        res = []
        while True:
            while root is not None:
                stack.append(root)
                root = root.left
            if not stack:
                break
            root = stack.pop()
            res.append(root.val)
            if root.right is not None:
                root = root.right
            else:
                root = None
        return res[-k]